components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -313,6 +325,70 @@ private void validateClaimUpdate(String username) throws UserStoreException {
         }
     }
 
+    /**
+     * Validate whether the updating groups do exist in the system.
+     *
+     * @param claimURI         Claim uri.
+     * @param value            Claim value.
+     * @param userStoreManager Userstore manager.
+     * @throws UserStoreException If the group does not exist in the system or if an error occurred while checking
+     *                            for group existence.
+     */
+    private void validateUserGroupClaim(String claimURI, String value, UserStoreManager userStoreManager)
+            throws UserStoreException {
+
+        Map<String, String> claimsMap = new HashMap<>();
+        claimsMap.put(claimURI, value);
+        validateUserGroups(claimsMap, userStoreManager);
+    }
+
+    /**
+     * Validate whether the updated groups does exist in the system.
+     *
+     * @param claims           List of claims to be updated.
+     * @param userStoreManager Userstore manager.
+     * @throws UserStoreException If the group does not exist in the system or if an error occurred while checking
+     *                            for group existence.
+     */
+    private void validateUserGroups(Map<String, String> claims, UserStoreManager userStoreManager)
+            throws UserStoreException {
+
+        if (claims == null || !claims.containsKey(GROUPS_LOCAL_CLAIM)) {
+            return;
+        }
+        /*
+         * We do not need to validate the groups for JIT provisioned users. That will be handled when resolved group
+         * mappings for the provisioned users. Therefore, this check can be skipped for the JIT provisioned users.
+         */
+        if (IdentityUtil.threadLocalProperties.get().get(FrameworkConstants.JIT_PROVISIONING_FLOW) != null &&
+                (Boolean) IdentityUtil.threadLocalProperties.get().get(FrameworkConstants.JIT_PROVISIONING_FLOW)) {
+            return;
+        }
+        String value = claims.get(GROUPS_LOCAL_CLAIM);
+        if (StringUtils.isBlank(value)) {
+            return;
+        }
+        // Resolve the multi attribute
+        String attributeSeparator =
+                userStoreManager.getRealmConfiguration().getUserStoreProperty(MULTI_ATTRIBUTE_SEPARATOR);
+        if (StringUtils.isEmpty(attributeSeparator)) {
+            attributeSeparator = DEFAULT_VALUE_SEPARATOR;
+        }
+        // We need to split if the user has provided a list of groups.
+        String[] groups = value.split(attributeSeparator);
+        boolean hasInvalidGroups = false;
+        for (String groupName : groups) {
+            // We need to identify the groups that does not exist in the system.
+            if (!userStoreManager.isExistingRole(groupName)) {
+                hasInvalidGroups = true;
```

Comment:
shall we add a debug log here logging the invalid group names.

Reply:
Fixed with a34f6b065f6b4f2316f6dce8ff05bb6262eae82f